medcat-service/medcat_service/main.py

```diff
@@ -1,28 +1,56 @@
+import logging
+from contextlib import asynccontextmanager
+
 import gradio as gr
 from fastapi import FastAPI, Request
 from fastapi.responses import JSONResponse
 
+from medcat_service.config import Settings
 from medcat_service.demo.gradio_demo import io
-from medcat_service.dependencies import get_settings
+from medcat_service.dependencies import set_global_processor
+from medcat_service.nlp_processor.medcat_processor import MedCatProcessor
 from medcat_service.routers import admin, health, process
 from medcat_service.types import HealthCheckFailedException
 
-settings = get_settings()
 
-app = FastAPI(
-    title="MedCAT Service",
-    summary="MedCAT Service",
-    contact={
+@asynccontextmanager
+async def lifespan(app: FastAPI):
+
+    log = logging.getLogger(__name__)
+    log.debug("Starting MedCAT Service lifespan setup")
+
+    # allow overriding settings and medcat processor for testing
+    settings = getattr(app.state, "settings", None)
+    if settings is None:
+        settings = Settings()
+        app.state.settings = settings
+
+    medcat = getattr(app.state, "medcat", None)
+    if medcat is None:
+        medcat = MedCatProcessor(settings)
+        app.state.medcat = medcat
+
+    app.state.settings = settings
+    app.state.medcat = medcat
+    app.state.title = "MedCAT Service",
+    app.state.summary = "MedCAT Service",
+    app.state.contact = {
         "name": "CogStack Org",
         "url": "https://cogstack.org/",
         "email": "contact@cogstack.org",
     },
-    license_info={
+    app.state.license_info = {
         "name": "Apache 2.0",
         "identifier": "Apache-2.0",
     },
-    root_path=settings.app_root_path,
-)
+    app.state.root_path = settings.app_root_path
+
+    set_global_processor(medcat)
```

Comment:
What about `set_global_settings`? Does that never get set?
I guess `get_global_settings` is never called either, so maybe we don't need that at all?